Error building KOReader for Kobo from source in Linux 17.3 (Ubuntu 14.04)

@houqp 
This is my third time trying to build KOReader. Maybe I should upgrade my Linux distro.
I have written what I did (some messages are in Spanish):

OS: Linux Mint 17.3 (it uses repositories of Ubuntu 14.04). Few software installed but everything is updated to the last version in the repositories. My computer is an i7 with 16 GB of RAM.
pkg-config-arm-linux-gnueabi and pkg-config-arm-linux-gnueabihf are not installed.

What I do:
In terminal I type:
sudo apt-get install build-essential git patch wget unzip \
gettext autoconf cmake libtool nasm luarocks \
libssl-dev libffi-dev libsdl2-dev libc6-dev-i386 xutils-dev linux-libc-dev:i386 zlib1g:i386
(Almost everything was already in their most recent version.)

Then I type in terminal:
sudo apt-get install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf
(But everything was already in their most recent version.)

Then I type:
sudo apt-get install ant
(openjdk-8-jdk is not in the repositories. I guess I didn't install ant and openjdk-8-jdk the other two times. The instructions on github.com/koreader/koreader are a bit confussing. I have installed openjdk-8-jdk through a ppa following the instructions here: http://ubuntuhandbook.org/index.php/2015/01/install-openjdk-8-ubuntu-14-04-12-04-lts/ )

Now I type:
git clone https://github.com/koreader/koreader.git
and
cd koreader && ./kodev fetch-thirdparty
(No problems so far that I can see)

But when I finally type:
./kodev release kobo
I get this:

> WARN:
>  Transifex client not found, no translation pulled.
> make -C base
> make[1]: se ingresa al directorio «/home/juan/koreader/base»
> install -d build/arm-linux-gnueabihf/libs
> install -d thirdparty/luajit/build/arm-linux-gnueabihf
> cd thirdparty/luajit/build/arm-linux-gnueabihf && \
> 		cmake -DCC="gcc" \
> 		-DXCOMPILE:BOOL=on \
> 		-DBASE_CFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer" -DHOST_CFLAGS="-march=native -O2 -ffast-math -pipe -fomit-frame-pointer " \
> 		 \
> 		-DTARGET_SONAME=libluajit.so \
> 		-DTARGET_CFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer  -march=armv7-a -mtune=cortex-a8 -mfpu=neon -mthumb -mfloat-abi=hard -fno-finite-math-only -fno-stack-protector -U_FORTIFY_SOURCE -fPIC -DKOBO_PLATFORM" \
> 		-DCROSS="arm-linux-gnueabihf-" \
> 		/home/juan/koreader/base/thirdparty/luajit && \
> 		make
> -- The C compiler identification is GNU 4.8.4
> -- The CXX compiler identification is GNU 4.8.4
> -- Check for working C compiler: /usr/bin/cc
> -- Check for working C compiler: /usr/bin/cc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Found Git: /usr/bin/git (found version "1.9.1") 
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf
> make[2]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[3]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[4]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> Scanning dependencies of target luajit
> make[4]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[4]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> [ 12%] Creating directories for 'luajit'
> [ 25%] Performing download step for 'luajit'
> Clonar en «luajit»...
> remote: Counting objects: 14191, done.
> remote: Total 14191 (delta 0), reused 0 (delta 0), pack-reused 14191
> Receiving objects: 100% (14191/14191), 4.62 MiB | 982.00 KiB/s, done.
> Resolving deltas: 100% (11615/11615), done.
> Comprobando la conectividad… hecho.
> Note: checking out 'v2.1.0-beta2'.
> 
> You are in 'detached HEAD' state. You can look around, make experimental
> changes and commit them, and you can discard any commits you make in this
> state without impacting any branches by performing another checkout.
> 
> If you want to create a new branch to retain commits you create, you may
> do so (now or later) by using -b with the checkout command again. Example:
> 
>   git checkout -b new_branch_name
> 
> HEAD se encuentra en 3e4a196... RELEASE LuaJIT-2.1.0-beta2
> [ 37%] No patch step for 'luajit'
> [ 50%] No update step for 'luajit'
> [ 62%] No configure step for 'luajit'
> [ 75%] Performing build step for 'luajit'
> make[5]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit»
> Building LuaJIT 2.1.0-beta2
> make[6]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> +--------------------------------------------------------------------------+
> | WARNING: Compiling the amalgamation needs a lot of virtual memory        |
> | (around 300 MB with GCC 4.x)! If you don't have enough physical memory   |
> | your machine will start swapping to disk and the compile will not finish |
> | within a reasonable amount of time.                                      |
> | So either compile on a bigger machine or use the non-amalgamated build.  |
> +--------------------------------------------------------------------------+
> make[7]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> HOSTCC    host/minilua.o
> HOSTCC    host/buildvm_asm.o
> HOSTCC    host/buildvm_peobj.o
> HOSTCC    host/buildvm_lib.o
> HOSTCC    host/buildvm_fold.o
> CC        luajit.o
> HOSTLINK  host/minilua
> /usr/bin/ld: se salta el /usr/lib/gcc/x86_64-linux-gnu/4.8/libgcc.a incompatible mientras se busca -lgcc
> /usr/bin/ld: no se puede encontrar -lgcc
> /usr/bin/ld: se salta el /usr/lib/gcc/x86_64-linux-gnu/4.8/libgcc_s.so incompatible mientras se busca -lgcc_s
> /usr/bin/ld: no se puede encontrar -lgcc_s
> collect2: error: ld returned 1 exit status
> make[7]: *** [host/minilua] Error 1
> make[7]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> make[6]: *** [amalg] Error 2
> make[6]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> make[5]: *** [amalg] Error 2
> make[5]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit»
> make[4]: *** [luajit-prefix/src/luajit-stamp/luajit-build] Error 2
> make[4]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[3]: *** [CMakeFiles/luajit.dir/all] Error 2
> make[3]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[2]: *** [all] Error 2
> make[2]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[1]: *** [build/arm-linux-gnueabihf/luajit] Error 2
> make[1]: se sale del directorio «/home/juan/koreader/base»
> make: *** [base/build/arm-linux-gnueabihf/luajit] Error 2
> juan@VALENTINA-M17 ~/koreader $ ^C
> juan@VALENTINA-M17 ~/koreader $ ./kodev release kobo
> WARN: Transifex client not found, no translation pulled.
> make -C base
> make[1]: se ingresa al directorio «/home/juan/koreader/base»
> install -d thirdparty/luajit/build/arm-linux-gnueabihf
> cd thirdparty/luajit/build/arm-linux-gnueabihf && \
> 		cmake -DCC="gcc" \
> 		-DXCOMPILE:BOOL=on \
> 		-DBASE_CFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer" -DHOST_CFLAGS="-march=native -O2 -ffast-math -pipe -fomit-frame-pointer " \
> 		 \
> 		-DTARGET_SONAME=libluajit.so \
> 		-DTARGET_CFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer  -march=armv7-a -mtune=cortex-a8 -mfpu=neon -mthumb -mfloat-abi=hard -fno-finite-math-only -fno-stack-protector -U_FORTIFY_SOURCE -fPIC -DKOBO_PLATFORM" \
> 		-DCROSS="arm-linux-gnueabihf-" \
> 		/home/juan/koreader/base/thirdparty/luajit && \
> 		make
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf
> make[2]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[3]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[4]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[4]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[4]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> [ 12%] Performing build step for 'luajit'
> make[5]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit»
> Building LuaJIT 2.1.0-beta2
> make[6]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> +--------------------------------------------------------------------------+
> | WARNING: Compiling the amalgamation needs a lot of virtual memory        |
> | (around 300 MB with GCC 4.x)! If you don't have enough physical memory   |
> | your machine will start swapping to disk and the compile will not finish |
> | within a reasonable amount of time.                                      |
> | So either compile on a bigger machine or use the non-amalgamated build.  |
> +--------------------------------------------------------------------------+
> make[7]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> HOSTLINK  host/minilua
> /usr/bin/ld: se salta el /usr/lib/gcc/x86_64-linux-gnu/4.8/libgcc.a incompatible mientras se busca -lgcc
> /usr/bin/ld: no se puede encontrar -lgcc
> /usr/bin/ld: se salta el /usr/lib/gcc/x86_64-linux-gnu/4.8/libgcc_s.so incompatible mientras se busca -lgcc_s
> /usr/bin/ld: no se puede encontrar -lgcc_s
> collect2: error: ld returned 1 exit status
> make[7]: *** [host/minilua] Error 1
> make[7]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> make[6]: *** [amalg] Error 2
> make[6]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit/src»
> make[5]: *** [amalg] Error 2
> make[5]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf/luajit-prefix/src/luajit»
> make[4]: *** [luajit-prefix/src/luajit-stamp/luajit-build] Error 2
> make[4]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[3]: *** [CMakeFiles/luajit.dir/all] Error 2
> make[3]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[2]: *** [all] Error 2
> make[2]: se sale del directorio «/home/juan/koreader/base/thirdparty/luajit/build/arm-linux-gnueabihf»
> make[1]: *** [build/arm-linux-gnueabihf/luajit] Error 2
> make[1]: se sale del directorio «/home/juan/koreader/base»
> make: *** [base/build/arm-linux-gnueabihf/luajit] Error 2
> 
>

@Frenzie Well, to be honest, I don't know what I am trying to accomplish. And as I said before, instructions here on github to build KOReader are at least confusing. They are not well structured and sometimes you don't know what you need to install for what system. I installed openjdk-8 believing that it was necessary and that i had missed that step last times. My system is 64 bits.
I have just tried ./kodev build and I get errors too: these ones

> 
> make -C base
> make[1]: se ingresa al directorio «/home/juan/koreader/base»
> install -d thirdparty/libjpeg-turbo/build/x86_64-linux-gnu
> cd thirdparty/libjpeg-turbo/build/x86_64-linux-gnu && \
> 		cmake -DCC="gcc" -DCXX="g++" -DCHOST="" \
> 		-DCPPFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer  -march=native -fPIC" -DLDFLAGS="-Wl,-O1 -Wl,--as-needed -static-libstdc++" \
> 		 \
> 		/home/juan/koreader/base/thirdparty/libjpeg-turbo && \
> 		make
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu
> make[2]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[3]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[4]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[4]: se sale del directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[4]: se ingresa al directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> [ 12%] Performing patch step for 'libjpeg-turbo'
> libtoolize: You should add the contents of the following files to `aclocal.m4':
> libtoolize:   `/usr/share/aclocal/libtool.m4'
> libtoolize:   `/usr/share/aclocal/ltoptions.m4'
> libtoolize:   `/usr/share/aclocal/ltversion.m4'
> libtoolize:   `/usr/share/aclocal/ltsugar.m4'
> libtoolize:   `/usr/share/aclocal/lt~obsolete.m4'
> libtoolize: Consider adding `AC_CONFIG_MACRO_DIR([m4])' to configure.ac and
> libtoolize: rerunning libtoolize, to keep the correct libtool macros in-tree.
> libtoolize: Consider adding `-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
> /bin/sh: 1: aclocal: not found
> make[4]: *** [libjpeg-turbo-prefix/src/libjpeg-turbo-stamp/libjpeg-turbo-patch] Error 127
> make[4]: se sale del directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[3]: *** [CMakeFiles/libjpeg-turbo.dir/all] Error 2
> make[3]: se sale del directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[2]: *** [all] Error 2
> make[2]: se sale del directorio «/home/juan/koreader/base/thirdparty/libjpeg-turbo/build/x86_64-linux-gnu»
> make[1]: *** [build/x86_64-linux-gnu/libs/libjpeg.so.8] Error 2
> make[1]: se sale del directorio «/home/juan/koreader/base»
> make: *** [all] Error 2
> Failed to build emulator! Try run with -v for more information.

--- a/README.md
+++ b/README.md
@@ -77,9 +77,10 @@ and cross compiling.
 Users of Debian and Ubuntu can install the required packages using:
 ```
 sudo apt-get install build-essential git patch wget unzip \
-gettext autoconf cmake libtool nasm luarocks \
+gettext autoconf automake cmake libtool nasm luarocks \
 libssl-dev libffi-dev libsdl2-dev libc6-dev-i386 xutils-dev linux-libc-dev:i386 zlib1g:i386
 ```
+That's all you need to get the emulator up and running with `./kodev build` and `./kodev run`.
 
 Cross compile toolchains are available for Ubuntu users through these commands:
 ```
@@ -139,43 +140,6 @@ cd koreader && ./kodev fetch-thirdparty
 Building, Running and Testing
 =============================
 
-For EReader devices (kindle, kobo, pocketbook, ubuntu-touch)
----------------------
-
-To build installable package for Kindle:
-```
-./kodev release kindle
-```
-
-To build installable package for Kobo:
-```
-./kodev release kobo
-```
-
-To build installable package for PocketBook:
-```
-./kodev release pocketbook
-```
-
-To build installable package for Ubuntu Touch
-```
-./kodev release ubuntu-touch
-```
-
-You may checkout our [nightlybuild script][nb-script] to see how to build a
-package from scratch.
-
-For Android devices
--------------------
-
-Make sure the "android" and "ndk-build" tools are in your PATH environment
-variable and the NDK variable points to the root directory of the Android NDK.
-
-Then, run this command to build installable package for Android:
-```
-./kodev release android
-```
-
 For emulating KOReader on Linux, Windows and Mac OSX
 -------------
 
@@ -227,6 +191,42 @@ make KOR_BASE=../koreader-base
 This will be handy if you are developing koreader-base and want to test your
 modifications with kroeader frontend. NOTE: only support relative path for now.
 
+For EReader devices (kindle, kobo, pocketbook, ubuntu-touch)
+---------------------
+
+To build installable package for Kindle:
+```
+./kodev release kindle
+```
+
+To build installable package for Kobo:
+```
+./kodev release kobo
+```
+
+To build installable package for PocketBook:
+```
+./kodev release pocketbook
+```
+
+To build installable package for Ubuntu Touch
+```
+./kodev release ubuntu-touch
+```
+
+You may checkout our [nightlybuild script][nb-script] to see how to build a
+package from scratch.
+
+For Android devices
+-------------------
+
+Make sure the "android" and "ndk-build" tools are in your PATH environment
+variable and the NDK variable points to the root directory of the Android NDK.
+
+Then, run this command to build installable package for Android:
+```
+./kodev release android
+```
 
 Translation
 ===========

